### PR TITLE
Feat like/dislike

### PR DESCRIPTION
**Pull request by amartya-rs**
In this PR:
- added like a video feature

Files to check: [review me](https://github.com/amartya-rs/hhTube/pull/4/commits/261e15b37e9847ccea433ee0a40abc261d5ee881)

Preview: https://deploy-preview-4--hh-tube.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -19,6 +25,10 @@ const App = () => {
                <Route path="/login" element={<LoginPage />} />
                <Route path="/signup" element={<SignUpPage />} />
             </Route>
+            //private routes
```
Comment (iRohitGaur) [inline src/App.js]: suggestion: this should be inside braces `{}`

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Gave a suggestion, please check

Good to merge 👍🏻

File: src/context/like-context.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   //remove a video from liked videos
+   const removeFromLike = (video) => {
+      getLikedVideos("likes", {
+         method: "delete",
+         url: `/api/user/likes/${video._id}`,
+         headers: {
+            accept: "*/*",
+            authorization: localStorage.getItem("token"),
+         },
+         data: {},
```
Comment (zoef-yt) [inline src/context/like-context.js]: You can remove this

File: src/context/like-context.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   //fetching liked videos
+   useEffect(() => {
+      getLikedVideos("likes", {
+         method: "get",
+         url: "/api/user/likes",
+         headers: {
+            accept: "*/*",
+            authorization: localStorage.getItem("token"),
+         },
+         data: {},
```
Comment (zoef-yt) [inline src/context/like-context.js]: You can remove this

File: src/context/like-context.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext, useEffect } from "react";
+import { useAxios } from "../utils/useAxios";
+
+const LikeContext = createContext();
+
+const LikeProvider = ({ children }) => {
+   const { response: likedVideos, apiCall: getLikedVideos } = useAxios();
```
Comment (zoef-yt) [inline src/context/like-context.js]: You can change the name of the API call as you are posting, deleting, and not just getting.

File: src/components/video-card/VideoCard.jsx [hunk 5]
```diff
@@ -1,9 +1,15 @@
 import "./video-card.css";
 import { MdOutlineWatchLater, MdPlaylistAdd } from "react-icons/md";
-import { AiOutlineLike } from "react-icons/ai";
+import { AiOutlineLike, AiFillLike } from "react-icons/ai";
+import { useLike } from "../../context/like-context";
+import { isPresentInData } from "../../utils/isPresentInData";
+import { useAuth } from "../../context/auth-context";
```
Comment (zoef-yt) [inline src/components/video-card/VideoCard.jsx]: you can import all the context from 1 import statement using index.js

**Review by zoef-yt — APPROVED** [review]
The UI looks clean. The code also looks good. left some suggestions you can check them out. You are good to merge.

File: src/App.js (hunk 1 above)
Comment (amartya-rs) [reply src/App.js]: Done.

File: src/components/video-card/VideoCard.jsx (hunk 5 above)
Comment (amartya-rs) [reply src/components/video-card/VideoCard.jsx]: Done.

File: src/context/like-context.js (hunk 3 above)
Comment (amartya-rs) [reply src/context/like-context.js]: Done.

File: src/context/like-context.js (hunk 2 above)
Comment (amartya-rs) [reply src/context/like-context.js]: Done.

File: src/context/like-context.js (hunk 4 above)
Comment (amartya-rs) [reply src/context/like-context.js]: Done.

**Pull request merged**